perfkitbenchmarker/providers/ibmcloud/ibmcloud_disk.py

```diff
@@ -0,0 +1,206 @@
+
+  def _FindVdiskIdByName(self, vdisks: [str], name: str):
+    """Finds vdisk id by name"""
+    for vdisk in vdisks:
+      if vdisk['name'] == name:
+        return vdisk['id']
+    return None
+
+  def Attach(self, vm):
+    """Attaches the disk to a VM.
+
+    Args:
+      vm: instance of the vm to which the disk will be attached.
+    """
+    if self.vol_id is None:
+      raise errors.Error(f'Cannot attach remote volume {self.name}')
+    if vm.vmid is None:
+      raise errors.Error(f'Cannot attach remote volume {self.name} '\
+                         'to non-existing {vm.name} VM')
+
+    volcmd = ibm.IbmAPICommand(self)
+    volcmd.flags.update({
+      'name': self.name,
+      'instanceid': vm.vmid,
+      'volume': self.vol_id,
+      'delete': True
+      })
+
+    # wait till volume is PROVISIONED before attach
+    status = None
+    endtime = time.time() + _MAX_DISK_ATTACH_SECONDS
+    while time.time() < endtime:
+      status = json.loads(volcmd.ShowVolume())['status']
+      logging.info('Checking volume status: %s', status)
+      if status == ibm.States.AVAILABLE:
+        logging.info('Volume is available')
+        break
+      time.sleep(2)
+
+    if status != ibm.States.AVAILABLE:
+      logging.error('Failed to create a volume')
+      raise errors.Error('IBMCLOUD ERROR: failed to provision a volume.')
+
+    resp = json.loads(volcmd.InstanceCreateVolume())
+    logging.info('Attached volume, volcmd.flags %s', volcmd.flags)
+    volume = resp['id']
+    logging.info('VDISK ID: %s, vdiskname: %s, instanceid: %s', volume, self.name, vm.vmid)
+    self.attached_vdisk_URI = volume
+    self.attached_vm = vm
+    self._WaitForVolumeAttachment(vm)
+    self._GetDeviceFromVDisk(vm)
+
+  @vm_util.Retry(poll_interval=1, max_retries=-1, timeout=_MAX_DISK_ATTACH_SECONDS, log_errors=False,
+                 retryable_exceptions=errors.Resource.RetryableCreationError)
+  def _WaitForVolumeAttachment(self, vm):
+    """Waits and checks until the volume status is attached """
+    if self.vol_id is None:
+      return
+    volcmd = ibm.IbmAPICommand(self)
+    volcmd.flags.update({
+      'instanceid': self.attached_vm.vmid,
+      'volume': self.vol_id
+      })
+    logging.info('Checking volume on instance %s, volume: %s', vm.vmid, self.vol_id)
+    status = None
+    endtime = time.time() + _MAX_DISK_ATTACH_SECONDS
+    while time.time() < endtime:
+      resp = volcmd.InstanceShowVolume()
+      if resp:
+        resp = json.loads(resp)
+        status = resp.get('status', 'unknown')
+        logging.info('Checking instance volume status: %s', status)
+        if status == ibm.States.ATTACHED:
+          logging.info('Remote volume %s has been attached to %s', self.name, vm.name)
+          break
+      time.sleep(2)
+    if status != ibm.States.ATTACHED:
+      logging.error('Failed to attach the volume')
+      raise errors.Error('IBMCLOUD ERROR: failed to attach a volume.')
+
+  def _GetDeviceFromVDisk(self, vm):
```

Comment:
Wrap this all in a 
  @vm_util.Retry(...)
decorator.  See this example:
  https://github.com/GoogleCloudPlatform/PerfKitBenchmarker/blob/4f74b8ebff3c18d53c289b0923600ef8a3728b98/perfkitbenchmarker/providers/gcp/gcp_cloud_redis.py#L121

Reply:
done.